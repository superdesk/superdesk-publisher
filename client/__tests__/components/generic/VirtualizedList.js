import React from "react";
import VirtualizedList from "../../../components/generic/VirtualizedList";
import { render } from "@testing-library/react";

jest.mock("react-virtualized/styles.css", () => {
  return {};
});

describe("generic/VirtualizedList", () => {
  const itemRendered = () => <div>test</div>;

  it("renders correctly", () => {
    const { container } = render(
      <div style={{ width: "500px", height: "500px" }}>
        <VirtualizedList
          hasNextPage={false}
          isNextPageLoading={false}
          loadNextPage={jest.fn()}
          items={[{}, {}]}
          ItemRenderer={itemRendered}
        />
      </div>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
