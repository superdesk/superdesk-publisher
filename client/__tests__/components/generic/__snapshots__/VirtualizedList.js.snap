// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic/VirtualizedList renders correctly 1`] = `
<div
  style="width: 500px; height: 500px;"
>
  <div
    style="overflow: visible; height: 0px; width: 0px;"
  >
    <div
      aria-label="grid"
      aria-readonly="true"
      class="ReactVirtualized__Grid ReactVirtualized__List"
      role="grid"
      style="box-sizing: border-box; direction: ltr; height: 0px; position: relative; width: 0px; will-change: transform; overflow-x: hidden; overflow-y: auto;"
      tabindex="0"
    />
  </div>
  <div
    class="resize-triggers"
  >
    <div
      class="expand-trigger"
    >
      <div
        style="width: 1px; height: 1px;"
      />
    </div>
    <div
      class="contract-trigger"
    />
  </div>
</div>
`;
